hbase-client/src/main/java/org/apache/hadoop/hbase/client/ConnectionImplementation.java

```diff
@@ -968,6 +974,19 @@ private RegionLocations locateRegionInMeta(TableName tableName, byte[] row, bool
     }
   }
 
+  void takeUserRegionLock() throws IOException {
+    try {
+      long waitTime = connectionConfig.getScannerTimeoutPeriod();
```

Comment:
Seems not push the latest commit?

Reply:
@infraio  I moved the acquiring of lock inside try catch block. https://github.com/apache/hbase/blob/58cb1cdaba1dbf952647934be0f34b4bad2e71db/hbase-client/src/main/java/org/apache/hadoop/hbase/client/ConnectionImplementation.java#L868
Also added a test case for that.

Reply:
Why move the takeUserRegionLock to inside "try catch block"? The right fix is to use operation timeout......

Reply:
I am -1 to use scanner timeout. It is too weird and confused. Operation timeout is not the best choice too but better. Thanks.

Reply:
> Reread RpcRetryingCallerImpl#callWithRetries code. If callable.prepare throw the LockTimeoutException, it will not retry. There are a check about call duration and callTimeout in line 156.

My point here is that your 15 seconds SLA case is not right. It is still meet your SLA even you use the operation timeout. I didn't mean that "move takeUserRegionLock  to try catch block". Thanks.

Reply:
> Operation timeout is not the best choice too but better

@infraio  In scan operation, there are 2 operations. One is to wait for lock and other is to wait for rpc to complete. On top of that we have retries. The problem we are trying to solve here is what is the timeout to use for lock. If we wait for operation timeout period and if it can't get the lock after the timeout, it will not have any time remaining for next attempts. So I am confused when you suggest to use operation timeout, are you suggesting to wait for operation timeout period while trying to get lock ?

Reply:
@saintstack  Could you please chime in with your inputs. I think we are going back and forth on which timeout to use. Also I have created https://issues.apache.org/jira/browse/HBASE-24983 to wrap the whole scan operation within operation timeout but is outside the scope of this jira. Thank you !

Reply:
> If we wait for operation timeout period and if it can't get the lock after the timeout, it will not have any time remaining for next attempts. 

Yes. The guarantee is that the operation will fail or success within the "operation timeout". No remaining time to retry and failed the operation is acceptable. 

> are you suggesting to wait for operation timeout period while trying to get lock 
Yes. Use the operation timeout period when wait for lock, instead of the scanner timeout now.

> I think we are going back and forth on which timeout to use.
I thought my point is clearly since we start this discussion. I suggested that use operation timeout instead of scanner timeout. Then you give me a 15 seconds SLA example. Then I checked the code: use operation timeout can meet your SLA requirements, too. So why not use operation timeout?

Reply:
@infraio  Thank you for your comment. Now I understand better what you mean. Let me update the PR by today. Thank you for being so patient with me.